tfhe/src/shortint/engine/mod.rs

```diff
@@ -114,7 +114,7 @@ where
             .iter_mut()
             .for_each(|a| {
                 let f_eval = f(i as u64);
-                *a = f_eval * delta;
+                *a = f_eval.wrapping_mul(delta);
```

Comment:
`shortint::key_switching_key::test::gen_multi_keys_test_truncate` fails on main with `attempt to multiply with overflow` (with overflow checks enabled)

But it's because of a bug which I'll push a fix for
I'll remove this wrapping_mul and add an assert `f_eval < message * carry`

I think we should enable overflow checks on CI tests

Reply:
yeah for the overflow checks it's in the backlog

what's the bug ?

but I think the code itself should do a checked mul so that we can warn/crash on overflow here as it means the table is likely broken

Reply:
Yes, the assert should be equivalent to a checked mul